tracer/test/snapshots/AspNetMvc4Tests.Integrated.WithExpansion.__path=_Home_badrequest-TransferRequest=true_statusCode=500.verified.txt

```diff
@@ -17,7 +17,8 @@
       http.method: GET,
       http.request.headers.host: localhost:00000,
       http.status_code: 500,
-      http.url: /home/badrequest?transferrequest=true,
+      http.url: http://localhost/Error/Index?errorId=Guid_2,
```

Comment:
I _think_ this is wrong, as it's using the wrong URL here. This stuff with TransferRequest is very tricky. But _maybe_ it was wrong before, and now it's right 😬 

@zacharycmontoya, what's your understanding of how all the spans in this snapshot should look? You worked on the transfer request stuff right, and it breaks my brain 😅

Reply:
As a quick summary, here's what happens in this request:
- The original HTTP request creates span aspnet.request with SpanId: Id_4
- Then it creates span aspnet-mvc.request with SpanId: Id_5
- Which then creates an error, so we run TransferRequest which internally starts a new HTTP request through the ASP.NET pipeline
- This new HTTP request creates span aspnet.request with SpanId: Id_3
- Then it creates span aspnet-mvc.request with SpanId: Id_2

The real differences in the snapshot file are for the http.url tags generated for the second internal HTTP request (which isn't user-facing). If we think it's valid to generate an http.url for these spans (should we?), then the new value seems correct because it matches what's captured in the resource and route tags.

Reply:
TLDR the `http.url` tag looks good to me. The `http.route` tag looks incorrect but I've introduced this in another comment so I'll resolve this thread.